src/NServiceBus.Core/Impersonation/Windows/WindowsIdentityEnricher.cs

```diff
@@ -19,11 +18,7 @@ public void MutateOutgoing(LogicalMessage logicalMessage, TransportMessage trans
                 return;
             }
             var windowsIdentity = WindowsIdentity.GetCurrent();
-            if (windowsIdentity != null)
```

Comment:
You linked to Thread.CurrentPrincipal. I changed the WindowsIdentity access. Sure we are talking about the same. Don't get me wrong I don't want to imply anything here, I'd just like to understand the issue and then happy to revert if necessary